### PR TITLE
Adding default endpoint for Elastic Rerank

### PR DESCRIPTION
**Pull request by ymao1**
Resolves https://github.com/elastic/ml-team/issues/1404

## Summary

Adds default endpoint for Elastic Rerank model. To verify, run locally and then access

```
GET http://localhost:9200/_inference/_all
```

The response should include the new default endpoint for the rerank model

```
{
    "inference_id": ".rerank-v1-elasticsearch",
    "task_type": "rerank",
    "service": "elasticsearch",
    "service_settings": {
        "num_threads": 1,
        "model_id": ".rerank-v1",
        "adaptive_allocations": {
            "enabled": true,
            "min_number_of_allocations": 0,
            "max_number_of_allocations": 32
        }
    },
    "task_settings": {
        "return_documents": true
    }
}
```

Then use the endpoint

```
POST http://localhost:9200/_inference/rerank/.rerank-v1-elasticsearch
{
  "query": "yo",
  "input": [
    "Hello world", "Goodnight moon"
  ]
}
```

The first time you call this API, it should download the model.

### CODE REVIEW

File: x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elasticsearch/ElasticsearchInternalService.java [hunk 1]
```diff
@@ -904,12 +917,19 @@ private List<Model> defaultConfigs(boolean useLinuxOptimizedModel) {
             ),
             ChunkingSettingsBuilder.DEFAULT_SETTINGS
         );
-        return List.of(defaultElser, defaultE5);
+        var defaultRerank = new ElasticRerankerModel(
```
Comment (ymao1) [inline x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elasticsearch/ElasticsearchInternalService.java:920]: Just checking that this should be the model we use and not `CustomElandRerankModel`

File: x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elasticsearch/ElasticsearchInternalService.java [hunk 2]
```diff
@@ -904,12 +917,19 @@ private List<Model> defaultConfigs(boolean useLinuxOptimizedModel) {
...
+            DEFAULT_RERANK_ID,
+            TaskType.RERANK,
+            NAME,
+            new ElasticRerankerServiceSettings(null, 1, RERANKER_ID, new AdaptiveAllocationsSettings(Boolean.TRUE, 0, 32)),
+            null
```
Comment (ymao1) [inline x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elasticsearch/ElasticsearchInternalService.java:925]: Seems like chunking is not supported for the rerank task type so I set this to `null`.

File: x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elasticsearch/ElasticsearchInternalService.java [hunk 3]
```diff
@@ -653,21 +662,24 @@ public void inferRerank(
     ) {
         var request = buildInferenceRequest(model.mlNodeDeploymentId(), new TextSimilarityConfigUpdate(query), inputs, inputType, timeout);
 
-        var modelSettings = (CustomElandRerankTaskSettings) model.getTaskSettings();
-        var requestSettings = CustomElandRerankTaskSettings.fromMap(requestTaskSettings);
-        Boolean returnDocs = CustomElandRerankTaskSettings.of(modelSettings, requestSettings).returnDocuments();
+        var returnDocs = Boolean.TRUE;
+        if (model instanceof CustomElandRerankModel) {
```
Comment (ymao1) [inline x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elasticsearch/ElasticsearchInternalService.java:666]: I was getting a `ClassCastException` here for the default `ElasticRerankerModel` so I'm doing this check. Do we need to support request level task settings for the default model as well?

**Review by davidkyle — COMMENTED** [review]
Looks good. 

I made a suggestion but if you want to do that in another PR that is fine

File: x-pack/plugin/inference/qa/inference-service-tests/src/javaRestTest/java/org/elasticsearch/xpack/inference/DefaultEndPointsIT.java [hunk 4]
```diff
@@ -125,6 +125,40 @@ private static void assertDefaultE5Config(Map<String, Object> modelConfig) {
         assertDefaultChunkingSettings(modelConfig);
     }
 
+    @SuppressWarnings("unchecked")
+    public void testInferDeploysDefaultRerank() throws IOException {
+        var model = getModel(ElasticsearchInternalService.DEFAULT_RERANK_ID);
+        assertDefaultRerankConfig(model);
```
[the remark this answers is not in the document]
Comment (ymao1) [reply x-pack/plugin/inference/qa/inference-service-tests/src/javaRestTest/java/org/elasticsearch/xpack/inference/DefaultEndPointsIT.java:134]: Updated in https://github.com/elastic/elasticsearch/pull/117939/commits/dc16082986b8f251052092c7ace63d291035af30

File: x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elasticsearch/ElasticsearchInternalService.java (hunk 3 above)
Comment (ymao1) [reply x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/elasticsearch/ElasticsearchInternalService.java]: Updated in https://github.com/elastic/elasticsearch/pull/117939/commits/dc16082986b8f251052092c7ace63d291035af30

**Review by davidkyle — APPROVED** [review]
LGTM

**Comment by davidkyle** [comment]
ping @szabosteve

**Pull request merged**